File: client/recipe-book/src/app/model/recipe.ts
import {Step} from './step';
import {Ingredient} from './ingredient';
import {Category} from './category';

export interface Recipe {
  id: number;
  title: string;
  cookingTime: number;
  steps: Step[];
  ingredients: Ingredient[];
  createdAt: Date;
  imgId?: string;
  description?: string;
  category: Category;
  tags: string[];
}

File: server/RecipeBook/src/main/java/ru/etu/recipebook/model/Recipe.java
package ru.etu.recipebook.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Recipe {
    @Id
    @GeneratedValue
    private long id;

    private String title;

    private int cookingTime;

    private LocalDateTime createdAt;

    private String imgId;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User createdBy;
}
